File: Tree/199. Binary Tree Right Side View.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> a = new ArrayList<Integer>();
    int level = 1;
    public List<Integer> rightSideView(TreeNode root) {
        if(root==null){
            return a;
        }
        if(a.size()<level){
            a.add(root.val);
        }else{
            a.set(level-1,root.val);
        }
        level++;
        rightSideView(root.left);
        rightSideView(root.right);
        level--;
        return a;
    }
}

File: Tree/199. Binary Tree Right Side View.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> a;
    int level = 1;
    vector<int> rightSideView(TreeNode* root) {
        if(root==nullptr){
            return a;
        }
        if(a.size()<level){
            a.push_back(root->val);
        }else{
            a[level-1]=root->val;
        }
        level++;
        rightSideView(root->left);
        rightSideView(root->right);
        level--;
        return a;
    }
};